source/common/modules/markdown-editor/plugins/render-codeblock-button.ts

```diff
@@ -0,0 +1,76 @@
+/**
+ * @ignore
+ * BEGIN HEADER
+ *
+ * Contains:        Codeblock copy button rendering plugin
+ * CVM-Role:        CodeMirror Plugin
+ * Maintainer:      Hendrik Erz
+ * License:         GNU GPL v3
+ *
+ * Description:     This plugin renders copy-to-clipboard button
+ *
+ * END HEADER
+ */
+
+import CodeMirror, { commands } from 'codemirror'
+
+const clipboard = window.clipboard
+/**
+   * Declare the  markdownRenderCodeBlockButton command
+   *
+   * @param   {CodeMirror.Editor}  cm  The CodeMirror instance
+*/
+;(commands as any).markdownRenderCodeBlockButton = function (cm: CodeMirror.Editor) {
+  cm.startOperation()
+
+  const codeBlockRE = /^(?:\s{0,3}`{3}|~{3}).*/
+  const lineCount = cm.lineCount()
+  let incodeblock = false
+  let codesblocks = new Array<string>()
+  let codeblock = ''
+  const codeblockClassOpen = 'code-block-first-line'
+  const codeblockClassClose = 'code-block-last-line'
+  const codeblocks = document.getElementsByClassName(codeblockClassOpen)
+
+  // This loop is grab the code block's code and store as a list
+  for (let j = 0; j < lineCount; j++) {
+    const line = cm.getLine(j)
+    if (codeBlockRE.test(line) && !incodeblock) {
+      codeblock = codeblock + cm.getLine(++j) + '\n'
+      incodeblock = true
+    } else if (codeBlockRE.test(line) && incodeblock) {
+      // Update the codes in code-block
+      // Remove newline character after the last code line
+      codeblock = codeblock.slice(0, -1)
+      codesblocks.push(codeblock)
+      codeblock = ''
+      incodeblock = false
+    } else if (incodeblock) {
+      codeblock = codeblock + cm.getLine(j) + '\n'
+    }
+  }
+  // This loop is update the copy buttons for each code block
+  if (document.getElementsByClassName(codeblockClassClose).length === codeblocks.length) {
+    for (let i = 0; i < codeblocks.length; i++) {
+      const codeBlock = codeblocks[i]
+      if (codeBlock === undefined) {
```

Comment:
This should never be true! Did you add this because you were encountering undefined-issues? If so then there is a serious bug further above in this file. If not, then it should be safe to delete this.

Reply:
@nathanlesage Yes, there will have some issues if I delete this one. I will take look into it to find a better solution.

Reply:
Hi @nathanlesage, should I delete line 53 or 56 or both?